scripts/generate_ipfs_hashes.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""
+This script generates the IPFS hashes for all packages.
+
+This script requires that you have IPFS installed:
+- https://docs.ipfs.io/guides/guides/install/
+"""
+
+import collections
+import csv
+import os
+import shutil
+import signal
+import subprocess  # nosec
+import sys
+import time
+from typing import Dict
+
+import ipfshttpclient
+
+from aea.helpers.ipfs.base import IPFSHashOnly
+
+AUTHOR = "fetchai"
+CORE_PATH = "aea"
+CORE_PACKAGES = {"connections": "stub", "protocols": "default", "skills": "error"}
+PACKAGE_PATH = "packages/fetchai"
+PACKAGE_TYPES = ["agents", "connections", "protocols", "skills"]
+PACKAGE_HASHES_PATH = "packages/hashes.csv"
+
+
+def ipfs_hashing(
+    package_hashes: Dict[str, str],
+    target_dir: str,
+    package_type: str,
+    package_name: str,
+    ipfs_hash_only: IPFSHashOnly,
+):
+    """Hashes a package and its components."""
+    print("Processing package {} of type {}".format(package_name, package_type))
+
+    if package_type != "agents":
+        # hash inner components (all `.py` files)
+        result_list = client.add(target_dir, pattern="*.py")
```

Comment:
Here we are hashing only Python modules. However, there might be other files of different format that are somehow part of the package.

Reply:
I think we don't need to hash files other than the `.py` files for now. The package folder itself will be hashed so it will include other files.